guides/advanced_guides/faceted_search.md

```diff
@@ -0,0 +1,169 @@
+      "release_date": 1552521600
+  },
+  ...
+]
+```
+
+You would declare faceted attributes as follows:
+
+```bash
+$ curl \
+  -X POST 'http://localhost:7700/indexes/movies/settings' \
+  --data '{
+      "attributesForFaceting": [
+          "director",
+          "genre"
+      ]
+  }'
+```
+
+## Querying On Faceted Attributes
+
+When performing a search, you can specify:
+
+### 1. The facets for which to retrieve the matching count
+
+`facets=[<facetName>, <facetName>, ...]`
+
+This attribute can take two values:
+
+- `[<facetName>, <facetName>, ...]` (Array of strings, defaults to `null`)
+
+  An array of strings that contains the facets for which to retrieve the matching count. The number of remaining candidates for each specified facet is returned. If a facet name doesn't exist, it will be ignored.
+
+- `["*"]`
+
+  The `*` character can also be used. In that case, a count for all facets is returned.
+
+If the `facets` parameter has been set, the returned results will contain two additional fields:
+
+- `facets`: The number of remaining candidates for each specified facet.
+
+- `exhaustiveFacetsCount`:
+  Returns `true` if this count is **exhaustive**.
+  Otherwise, returns `false` if this count is **approximative**.
+
+### Example
+
+Given a movie database, suppose that you want to know how many Batman movies are classified as `Action and Adventure`, how many of them are classified as `Animation` and what the number of Batman movies per director is. You would use the following CURL command:
+
+```bash
+$ curl --get 'http://localhost:7700/indexes/movies/search' \
+    --data-urlencode 'q=Batman' \
+    --data-urlencode 'facets=["genre", "director"]'
+```
+
+### 2. The facets to filter on
+
+`facetFilters=["facetName:facetValue"]`, `facetFilters=[["facetName:facetValue"]]` or a mix of both `facetFilters=["facetName1:facetValue1", ["facetName2:facetValue2"]]`
+
+- `["facetName1:facetValue1", ["facetName2:facetValue2"]]` (Array of strings or arrays of strings, defaults to `null`)
+
+  Both types of array contain the facet names and values to filter on.
+  A valid array must be an array which contains either a list of strings or arrays of strings and can mix both (e.g. `["kind:t-shirt", ["color:red", "color:green"]]`).
+
+  - `facetName`: The name (the attribute) of a field used as a facet (e.g. `color`, `kind`).
+  - `facetValue`: The value of this facet to filter results on (e.g. `red`, `green`, `t-shirt`, `pants`).
+
+Inputting a double dimensional array allows you to use **logical connectives**.
+
+- **Inner arrays elements** are connected by an `OR` operator (e.g. `[["color:red", "color:green"]]`).
+- **Outer arrays elements** are connected by an `AND` operator (e.g. `["color:red", "kind:t-shirt"]`).
+
+You can mix connectives, for instance, the following array:
+
+```json
+[["genre:Horror", "genre:Comedy"], "director:Jordan Peele"]
+```
+
+Can be translated as:
```

Comment:
I thinks this is pretty confusing, I would maybe introduce the way it works, before the syntax.

Reply:
Don't hesitate to tell me if you want to modify/add things! :smile: